user-service/src/main/java/org/orcid/user/service/UserService.java

```diff
@@ -42,396 +42,401 @@
-		return userRepository.findByDeletedFalse(pageable).map(UserDTO::valueOf);
-	}
-
-	public Optional<User> getUserWithAuthoritiesByLogin(String login) {
-		return userRepository.findOneByLogin(login);
-	}
-
-	public Optional<User> getUserWithAuthorities(String id) {
-		return userRepository.findById(id);
-	}
-
-	public Optional<User> getUserWithAuthorities() {
-		return SecurityUtils.getCurrentUserLogin().flatMap(userRepository::findOneByLogin);
-	}
-
-	public List<User> findAllByLoginOrEmail(String login, String email) {
-		return userRepository.findAllByLoginOrEmail(login, email);
-	}
-
-	public void removeAuthorityFromUser(String id, String authority) {
-		Optional<User> existing = getUserWithAuthorities(id);
-		if (!existing.isPresent()) {
-			throw new BadRequestAlertException("User not present " + id, "user", null);
-		}
-
-		User user = existing.get();
-		if (user.getAuthorities() != null && !user.getAuthorities().isEmpty()) {
-			user.setAuthorities(
-					user.getAuthorities().stream().filter(a -> !a.equals(authority)).collect(Collectors.toSet()));
-		}
-		userRepository.save(user);
-	}
-	
-	public void removeOwnershipFromUser(String id) {
-            Optional<User> existing = getUserWithAuthoritiesByLogin(id);
-            if (!existing.isPresent()) {
-                    throw new BadRequestAlertException("User not present " + id, "user", null);
+            user.setAuthorities(getAuthoritiesForUser(userDTO, userDTO.getIsAdmin()));
+            if (!StringUtils.equals(user.getEmail(), userDTO.getEmail().toLowerCase())) {
+                user.setEmail(userDTO.getEmail().toLowerCase());
+                user.setActivated(false);
+                user.setActivationKey(RandomUtil.generateResetKey());
+                user.setActivationDate(Instant.now());
+                mailService.sendActivationEmail(user);
+            }
+            if (user.getSalesforceId() != null && userDTO.getSalesforceId() != null && !user.getSalesforceId().equals(userDTO.getSalesforceId())) {
+                user.setSalesforceId(userDTO.getSalesforceId());
+                user.setLastModifiedBy(SecurityUtils.getCurrentUserLogin().get());
+                user.setLastModifiedDate(Instant.now());
             }
 
-            User user = existing.get();
-            if (user.getAuthorities() != null && !user.getAuthorities().isEmpty()) {
-                    user.setAuthorities(
-                                    user.getAuthorities().stream().filter(a -> !a.equals(AuthoritiesConstants.ORG_OWNER)).collect(Collectors.toSet()));
+            userRepository.save(user);
+            userCaches.evictEntryFromUserCaches(user.getEmail());
+            userCaches.evictEntryFromUserCaches(user.getLogin());
+            LOG.debug("Changed Information for User: {}", user);
+            return user;
+        }).map(UserDTO::valueOf);
+    }
+
+    // public Optional<UserDTO> updateUserSalesForceId(UserDTO userDTO, String
```

Comment:
@Camelia-Orcid could you please remove this commented code?

Reply:
Removed.